event_model/__init__.py

```diff
@@ -453,6 +480,29 @@ def get_default_handler_cache():
     def inplace(self):
         return self._inplace
 
+    def register_handler(self, spec, handler, overwrite=False):
+        if (not overwrite) and (spec in self._handler_registry):
+            original = self._handler_registry[spec]
+            if original is handler:
+                return
+            raise DuplicateHandler(
+                f"There is already a handler registered for the spec {spec!r}."
```

Comment:
```suggestion
                f"There is already a handler registered for the spec {spec!r}. "
```

Reply:
Hmm, I had meant to commit this. Applied manually.

Reply:
No worries, glad it's all in.